src/meta/src/manager/catalog/mod.rs

```diff
@@ -403,48 +487,52 @@ where
                     table.name, ref_count
                 ))),
                 None => {
+                    let dependent_relations = table.dependent_relations.clone();
                     let mut transaction = Transaction::default();
-                    let mut users_need_update = Self::release_privileges(
+
+                    let mut tables_to_drop =
+                        future::join_all(internal_table_ids.into_iter().map(|id| async move {
+                            Table::select(self.env.meta_store(), &id).await
+                        }))
+                        .await
+                        .into_iter()
+                        .map_ok(|table| table.unwrap())
```

Comment:
`internal_table_ids` is passed from table fragments, this might include source internal tables which don't have any catalogs yet. You can ignore them for now and add a `FIXME` here and resolve it after source internal table get supported.